src/shared/fs/watchedFiles.ts

```diff
@@ -275,26 +281,44 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
     }
 
     /**
-     * Rebuilds registry using current glob and exclusion patterns.
-     * All functionality is currently internal to class, but can be made public if we want a manual "refresh" button
+     * Builds/rebuilds registry using current glob and exclusion patterns. ***Necessary to init registry***.
+     *
+     * @param cancellationPromise Cancels additional registrations when rejected. Otherwise, this
      */
-    public async rebuild(): Promise<void> {
+    public async rebuild(cancellationPromise: Promise<void>): Promise<void> {
+        let isCanceled = false
+        let skips = 0
+        this.isWatching = true
         this.reset()
 
         const exclude = getExcludePatternOnce()
-        for (const glob of this.globs) {
+        getLogger().info(`Building registry with the following criteria: ${this.outputPatterns()}`)
+        cancellationPromise.then(
+            () => {},
+            () => {
+                isCanceled = true
+            }
+        )
+        for (let i = 0; i < this.globs.length && !isCanceled; i++) {
+            const glob = this.globs[i]
             try {
                 const found = await vscode.workspace.findFiles(glob, exclude)
-                for (const item of found) {
-                    await this.addItem(item, true)
+                for (let j = 0; j < found.length && !isCanceled; j++) {
+                    await this.addItem(found[j], true)
                 }
             } catch (e) {
+                // file not processable
+                skips++
                 const err = e as Error
-                if (err.name !== 'Canceled') {
-                    getLogger().error('watchedFiles: findFiles("%s", "%s"): %s', glob, exclude, err.message)
-                }
+                getLogger().error('watchedFiles: findFiles("%s", "%s"): %s', glob, exclude, err.message)
             }
         }
+        getLogger().info(
+            'Registered %s items%s%s',
```

Comment:
other log messages in this module have a `${this.name}:` prefix

Reply:
The original didn't have it. Added.